diff --git a/vhci/vhci.c b/vhci/vhci.c
index 2acc514..618ddc3 100644
--- a/vhci/vhci.c
+++ b/vhci/vhci.c
@@ -4,6 +4,7 @@
 #include <linux/usb.h>
 #include <linux/usb/hcd.h>
 #include <linux/module.h>
+#include <linux/version.h>

 static dev_t bce_vhci_chrdev;
 static struct class *bce_vhci_class;
@@ -46,6 +47,9 @@ int bce_vhci_create(struct bce_device *dev, struct bce_vhci *vhci)
         goto fail_hcd;
     }
     vhci->hcd->self.sysdev = &dev->pci->dev;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,4,0)
+    vhci->hcd->self.uses_dma = 1;
+#endif
     *((struct bce_vhci **) vhci->hcd->hcd_priv) = vhci;
     vhci->hcd->speed = HCD_USB2;

@@ -695,7 +699,11 @@ static const struct hc_driver bce_vhci_driver = {
         .product_desc = "BCE VHCI Host Controller",
         .hcd_priv_size = sizeof(struct bce_vhci *),

+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,4,0)
+        .flags = HCD_USB2,
+#else
         .flags = HCD_USB2 | HCD_DMA,
+#endif

         .start = bce_vhci_start,
         .stop = bce_vhci_stop,
@@ -743,4 +751,4 @@ void __exit bce_vhci_module_exit(void)
 }

 module_param_named(vhci_port_mask, bce_vhci_port_mask, ushort, 0444);
-MODULE_PARM_DESC(vhci_port_mask, "Specifies which VHCI ports are enabled");
\ No newline at end of file
+MODULE_PARM_DESC(vhci_port_mask, "Specifies which VHCI ports are enabled");
